diff --git a/Makefile.mvs b/Makefile.mvs
new file mode 100644
index 0000000..301b460
--- /dev/null
+++ b/Makefile.mvs
@@ -0,0 +1,69 @@
+# gnu make required.  flex not required unless you change the .l file.
+
+.SUFFIXES: .maclib .macro .assemble .listing
+.PHONY: m
+
+%.assemble: %.s
+	./gas2asm -l <$< | ./asmxpnd >$@
+
+%.listing: %.assemble
+	hlasm -option term -macrobase macros $*
+
+INSTALLDIR?=install
+INSTALLBIN:=${INSTALLDIR}/bin
+INSTALLMAC:=${INSTALLDIR}/macros
+MACDIR:=macros
+MACREADME:=${MACDIR}/README.md
+MACROS:=${notdir ${MACS}}
+
+CFLAGS:=-g -Wall -Werror
+YFLAGS:=-Wall -Werror
+
+${INSTALLBIN}/%: %
+	cp -p $< $@
+
+${INSTALLMAC}/%: macros/%
+	cp -p $< $@
+
+M:=gas2asm asmxpnd
+
+all: ${M} ${MACREADME}
+
+${MACREADME}:
+	../bin/genmaclib ${MACDIR} '*.macro'
+	touch ${MACREADME}
+
+
+# Ensure that the generated flex output stays.	OpenSuse 11 ships a
+# broken flex that mangles things somewhat royal.
+gas2asm.c: gas2asm.l
+gas2asm.o: gas2asm.c symtab.c loadfile.c le128.c lookup3.c gas2asmmain.c
+
+${INSTALLBIN}:
+	mkdir -p ${INSTALLBIN}
+
+${INSTALLMAC}:
+	mkdir -p ${INSTALLMAC}
+
+install: ${M} ${MACREADME} ${INSTALLBIN} ${INSTALLMAC}
+	cp -f gas2asm asmxpnd ${INSTALLBIN}
+	cp -rpf ${MACDIR} ${INSTALLDIR}
+
+MP:=${addprefix ${INSTALLMAC}/,${MACROS}}
+install-macros: ${MP} | ${INSTALLMAC}
+
+t: $M
+	rm -f core
+	./gas2asm -f <t.s >t.out 2>t.dbg
+
+test: $M
+	rm -f core
+	./gas2asm -f <test/test.s >test.assemble 2>test.dbg
+	hlasm --macrobase macros test
+
+clean:
+	rm -f core *.o $M t.out t.dbg ${MACREADME}
+	make -f Makefile.mvs -C test clean
+
+check: $M
+	(cd test; make -f Makefile.mvs)
