diff --git a/test/Makefile.mvs b/test/Makefile.mvs
new file mode 100644
index 0000000..136d5f2
--- /dev/null
+++ b/test/Makefile.mvs
@@ -0,0 +1,30 @@
+.SUFFIXES: .assemble .text
+.PHONY: all clean check ok
+.SECONDARY: .assemble
+
+OLD_SHELL:=${SHELL}
+# SHELL=$(warning [target: $@  modified: ($?)])$(OLD_SHELL)
+MOD := ../gas2asm
+XPD := ../asmxpnd
+
+FNS:=${basename ${wildcard *.s}}
+ASM:=${addsuffix .assemble,${FNS}}
+
+%.assemble: %.s ${MOD} ${XPD}
+	((${MOD} <$< | ${XPD} >$@) || (mv $@ $*.asm; exit 1))
+
+check: ${ASM}
+	cat *.assemble >check.asm
+	diff check.asmok check.asm >check.out
+	@echo '>>> all pass <<<'
+
+ok:
+	mv check.asm check.asmok
+
+# This works only if you have a hlasm that runs on the platform
+# --macrobase should point to the directory containing the @file etc. macros
+%.text: %.assemble
+	hlasm --macrobase ../macros $*
+
+clean:
+	rm -f *.assemble *.text check.asm
