diff --git a/test/Makefile.mvs b/test/Makefile.mvs
new file mode 100644
index 0000000..a141e39
--- /dev/null
+++ b/test/Makefile.mvs
@@ -0,0 +1,23 @@
+.SUFFIXES: .assemble .o
+.PHONY: all clean check
+.SECONDARY: .assemble
+
+MOD := ../gas2asm
+XPD := ../asmxpnd
+GASDIR=.
+HLASMDIR=actual
+
+gassrc := $(shell find $(GASDIR) -type f -name '*.s')
+hlasmsrc := $(patsubst $(GASDIR)/%,$(HLASMDIR)/%,$(gassrc:.s=.assemble))
+
+$(HLASMDIR)/%.assemble: $(GASDIR)/%.s ${MOD} ${XPD}
+	((${MOD} <$< | ${XPD} >$@) || (mv $@ $*.asm; exit 1))
+
+check: $(hlasmsrc)
+	../../bin/split-check expected check.asmok
+# use xargs below to trim blanks around wc -l
+	fail=$(shell diff -qr actual expected | wc -l | xargs)
+	total=$(shell diff -qrs actual expected | wc -l | xargs)
+
+clean:
+	rm -f actual expected check.asm
